### PR TITLE
Initial infraestructure refactor

### PR DESCRIPTION
**Pull request by JesusPoderoso**
- API pending refactor
- New Exception and its usage
- Util classes renamed

### CODE REVIEW

File: lib/include/fastdds_qos_profiles_manager/transport_descriptor/TransportDescriptor.hpp [hunk 1]
```diff
@@ -1057,29 +964,16 @@ FASTDDS_QOS_PROFILES_MANAGER_DllAPI void clear_logical_port_increment(
...
-        const std::string& xml_file,
-        const std::string& transport_descriptor_id);
-
 /**
  * @brief Remove specific listening port from specific Transport Descriptor profile.
  *
  * @param[in] xml_file Absolute/relative path to the XML file.
  * @param[in] transport_descriptor_id Transport descriptor profile identifier.
- * @param[in] index Collection element to be removed.
+ * @param[in] index Collection element to be printed. If empty, the complete collection is erased.
```
Comment (JesusPoderoso) [inline lib/include/fastdds_qos_profiles_manager/transport_descriptor/TransportDescriptor.hpp:972]:
```suggestion
 * @param[in] index Collection element to be removed. If empty, the complete collection is erased.
```

File: lib/include/fastdds_qos_profiles_manager/transport_descriptor/TransportDescriptor.hpp [hunk 2]
```diff
@@ -1178,89 +1072,50 @@ FASTDDS_QOS_PROFILES_MANAGER_DllAPI void clear_tls_verify_file(
...
-        const std::string& xml_file,
-        const std::string& transport_descriptor_id);
-
 /**
  * @brief Remove specific TLS verification mode from specific Transport Descriptor profile mask.
  *
  * @param[in] xml_file Absolute/relative path to the XML file.
  * @param[in] transport_descriptor_id Transport descriptor profile identifier.
- * @param[in] index Collection element to be removed.
+ * @param[in] index Collection element to be printed. If empty, the complete collection is erased.
```
Comment (JesusPoderoso) [inline lib/include/fastdds_qos_profiles_manager/transport_descriptor/TransportDescriptor.hpp:1080]:
```suggestion
 * @param[in] index Collection element to be removed. If empty, the complete collection is erased.
```

File: lib/include/fastdds_qos_profiles_manager/transport_descriptor/TransportDescriptor.hpp [hunk 3]
```diff
@@ -1178,89 +1072,50 @@ FASTDDS_QOS_PROFILES_MANAGER_DllAPI void clear_tls_verify_file(
...
-        const std::string& xml_file,
-        const std::string& transport_descriptor_id);
-
 /**
  * @brief Remove specific TLS supported feature from specific Transport Descriptor profile mask.
  *
  * @param[in] xml_file Absolute/relative path to the XML file.
  * @param[in] transport_descriptor_id Transport descriptor profile identifier.
- * @param[in] index Collection element to be removed.
+ * @param[in] index Collection element to be printed. If empty, the complete collection is erased.
```
Comment (JesusPoderoso) [inline lib/include/fastdds_qos_profiles_manager/transport_descriptor/TransportDescriptor.hpp:1097]:
```suggestion
 * @param[in] index Collection element to be removed. If empty, the complete collection is erased.
```

File: lib/include/fastdds_qos_profiles_manager/transport_descriptor/TransportDescriptor.hpp [hunk 4]
```diff
@@ -1178,89 +1072,50 @@ FASTDDS_QOS_PROFILES_MANAGER_DllAPI void clear_tls_verify_file(
...
-        const std::string& xml_file,
-        const std::string& transport_descriptor_id);
-
 /**
  * @brief Remove specific TLS verification path from specific Transport Descriptor profile mask.
  *
  * @param[in] xml_file Absolute/relative path to the XML file.
  * @param[in] transport_descriptor_id Transport descriptor profile identifier.
- * @param[in] index Collection element to be removed.
+ * @param[in] index Collection element to be printed. If empty, the complete collection is erased.
```
Comment (JesusPoderoso) [inline lib/include/fastdds_qos_profiles_manager/transport_descriptor/TransportDescriptor.hpp:1114]:
```suggestion
 * @param[in] index Collection element to be removed. If empty, the complete collection is erased.
```

File: lib/src/cpp/utils/XMLManager.hpp [hunk 5]
```diff
@@ -207,6 +199,14 @@ class ParseXML
     std::unique_ptr<std::vector<uint>>  get_real_index(
             xercesc::DOMNodeList*& node_list);
 
+    /**
```
Comment (JesusPoderoso) [inline lib/src/cpp/utils/XMLManager.hpp:202]: This was no longer public, so moved to private method.

File: lib/include/fastdds_qos_profiles_manager/domain_participant/DomainParticipant.hpp [hunk 6]
```diff
@@ -422,29 +390,16 @@ FASTDDS_QOS_PROFILES_MANAGER_DllAPI void clear_participant_id(
...
-        const std::string& xml_file,
-        const std::string& profile_id);
-
 /**
  * @brief Remove specific user transport from specific Domain Participant profile.
  *
  * @param[in] xml_file Absolute/relative path to the XML file.
  * @param[in] profile_id Domain participant profile identifier.
- * @param[in] index Collection element to be removed.
+ * @param[in] index Collection element to be printed. If empty, the complete collection is erased.
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply lib/include/fastdds_qos_profiles_manager/domain_participant/DomainParticipant.hpp:398]: True. Copy/paste has its risks :sweat_smile:

File: lib/src/cpp/utils/ErrorHandlerXMLManager.hpp [hunk 7]
```diff
@@ -16,8 +16,8 @@
  * @file
  */
 
-#ifndef _FAST_DDS_QOS_PROFILES_MANAGER_UTILS_PARSE_XML_ERROR_HANDLER_HPP_
-#define _FAST_DDS_QOS_PROFILES_MANAGER_UTILS_PARSE_XML_ERROR_HANDLER_HPP_
+#ifndef _FAST_DDS_QOS_PROFILES_MANAGER_UTILS_XML_MANAGER_ERROR_HANDLER_HPP_
```
Comment (JLBuenoLopez-eProsima) [inline lib/src/cpp/utils/ErrorHandlerXMLManager.hpp:19]: You have altered the order. It should be:

```suggestion
#ifndef _FAST_DDS_QOS_PROFILES_MANAGER_UTILS_ERROR_HANDLER_XML_MANAGER_HPP_
```

File: lib/src/cpp/utils/TagsXMLManager.hpp [hunk 8]
```diff
@@ -16,8 +16,8 @@
  * @file
  */
 
-#ifndef _FAST_DDS_QOS_PROFILES_MANAGER_UTILS_PARSE_XML_TAGS_HPP_
-#define _FAST_DDS_QOS_PROFILES_MANAGER_UTILS_PARSE_XML_TAGS_HPP_
+#ifndef _FAST_DDS_QOS_PROFILES_MANAGER_UTILS_XML_MANAGER_TAGS_HPP_
```
Comment (JLBuenoLopez-eProsima) [inline lib/src/cpp/utils/TagsXMLManager.hpp:19]: Same comment again.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Pull request merged**